app/src/main/java/com/duckduckgo/app/browser/DownloadConfirmationFragment.kt

```diff
@@ -24,46 +24,58 @@ import android.view.View
 import android.view.ViewGroup
 import android.widget.Toast
 import androidx.core.content.FileProvider.getUriForFile
-import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.DownloadFileData
-import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.UserDownloadAction
+import com.duckduckgo.app.browser.downloader.FileDownloader
+import com.duckduckgo.app.browser.downloader.guessFileName
 import com.duckduckgo.app.global.view.gone
 import com.duckduckgo.app.global.view.leftDrawable
 import com.duckduckgo.app.global.view.show
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import kotlinx.android.synthetic.main.download_confirmation.view.*
 import timber.log.Timber
+import java.io.File
+import java.io.IOException
+import kotlin.concurrent.thread
 
 class DownloadConfirmationFragment(
-    private val downloadFileData: DownloadFileData,
-    private val userDownloadAction: UserDownloadAction
+    private val pendingDownload: FileDownloader.PendingFileDownload,
+    private val downloader: FileDownloader,
+    private val downloadListener: FileDownloader.FileDownloadListener,
+    private var file: File? = null
```

Comment:
do we need this file here? Seems like the value will be assigned inside `setupFile()`

Reply:
No not needed, glad you picked up on it. I had considered passing the File in and then moved to the setupFile() approach so there is no need for it to be exposed in the constructor.